vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -72,6 +78,35 @@
     private Registration validationRegistration;
     private Registration dataProviderListenerRegistration;
 
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(ListDataProvider<T> dataProvider) {
+        this.setDataProvider((DataProvider<T, ?>) dataProvider);
+        return getListDataView();
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(Collection<T> items) {
+        return setDataProvider(DataProvider.ofCollection(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(Stream<T> items) {
+        return setDataProvider(DataProvider.fromStream(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(T... items) {
+        return setDataProvider(DataProvider.ofItems(items));
+    }
```

Comment:
```
    @Override
    public CheckboxGroupListDataView<T> setDataProvider(ListDataProvider<T> dataProvider) {
        this.setDataProvider((DataProvider<T, ?>) dataProvider);
        return getListDataView();
    }

    @Override
    public CheckboxGroupListDataView<T> setDataProvider(Collection<T> items) {
        return setDataProvider(DataProvider.ofCollection(items));
    }

    @Override
    public CheckboxGroupListDataView<T> setDataProvider(Stream<T> items) {
        return setDataProvider(DataProvider.fromStream(items));
    }

    @Override
    public CheckboxGroupListDataView<T> setDataProvider(T... items) {
        return setDataProvider(DataProvider.ofItems(items));
    }
```
I think all of these methods could and should be added as default implementations to the list data view instead ?

Reply:
Three methods have been moved to `flow-data` except this one:

```
@Override
    public CheckboxGroupListDataView<T> setDataProvider(ListDataProvider<T> dataProvider) {
        this.setDataProvider((DataProvider<T, ?>) dataProvider);
        return getListDataView();
    }
```

because it still uses some component-specific code.

Reply:
It could be moved too if we expose a `setDataProvider(DataProvider dp, F initialFilter)` into `DataController`, it is already in `DataCommunicator`. But then we would need to also add `DataController getDataController()` to `DataView`. 

I think we can do that later if we decide so, to get forward now.